src/pages/projectSettings/Tabs.tsx

```diff
@@ -204,7 +204,7 @@ export const ProjectSettingsTabs: React.VFC<Props> = ({
         />
         <Route
           path={ProjectSettingsTabRoutes.EventLog}
-          element={<EventLogTab projectType={projectType} />}
+          element={<EventLogTab projectType={projectType} key={identifier} />}
```

Comment:
do we need the `key` here?

Reply:
It's a pretty straightforward way to make sure all of the state gets reset when navigating between projects as you flagged in your first comment. Updating the cache policy helped somewhat but things like the `allEventsFetched` state in our hook would not necessarily update.